### ISSUE TITLE
compiletest: what does `--pass` (pass mode) actually intend to do?

### ISSUE THREAD

**Issue by jieyouxu** [rust-lang/rust#131990]
Apparently compiletest has a `--pass` flag which allows setting "pass mode" but it's unclear what that is intended to do. AFAIK, many tests don't account for this at all, and `//@ ignore-pass` is apparently a directive which is used liberally. CI doesn't seem to have fanatastic coverage for `--pass` configurations either. It might also be a specific thing for a subset of test suites. And also, how does the current implementation compare to the intended behavior / use case.

See https://rust-lang.zulipchat.com/#narrow/channel/182449-t-compiler.2Fhelp/topic/.E2.9C.94.20CI-only.20failure.20in.20new.20test/near/477973621 for some discussions.

**Comment by Zalathar** [comment]
For archaeological reference, `--pass` and `//@ ignore-pass` appear to have been added in #61755.

**Comment by jieyouxu** [comment]
That PR itself is builtin upon https://github.com/rust-lang/rust/pull/61778

**Comment by jieyouxu** [comment]
TL;DR: pass mode seems to be a mechanism to allow someone to forcefully modify how deep into the compilation or test running pipeline (check vs build vs run) the tests will be evaluated at. `//@ ignore-pass` is the opt-out, and this seems to be used as an escape hatch because PR CI I think uses `--pass=check` to force downgrade some ui tests to run them at a lower test running pipeline to save CI time?

**Comment by ehuss** [comment]
In practice, I think `--pass=check` is the only mode override used. If you are not doing codegen work, it makes the ui tests much faster. (At least, that's how I documented it.) I'm not sure how useful the other modes are.
